scripts/actions/remove-completed-batch.js

```diff
@@ -60,32 +60,35 @@ const removePageContext = async (fileUris) => {
       };
     });
 
-  return contextUids.reduce(async (returnCode, { contextUid, fileUri }) => {
-    const url = new URL(
-      `/context-api/v2/projects/${PROJECT_ID}/contexts/${contextUid}`,
-      process.env.TRANSLATION_VENDOR_API_URL
-    );
-
-    const options = {
-      method: 'DELETE',
-      headers: {
-        Authorization: `Bearer ${accessToken}`,
-      },
-    };
-
-    const resp = await fetch(url.href, options);
-
-    const { response } = await resp.json();
-    const { code } = response;
-
-    if (code === 'SUCCESS' && resp.ok) {
-      console.log(`[*] Successfully deleted ${fileUri} context.`);
-    } else {
-      console.error(`[!] Unable to delete ${fileUri} context.`);
-      return code;
-    }
-    return returnCode;
-  }, 'SUCESSS');
+  return await contextUids.reduce(
```

Comment:
Since the value returned here is not a promise, but an array of promises, this `await` might return too early. You'll want to wrap this in a `Promise.all` so that you ensure all promises are resolved/rejected before the array of values is fully resolved.

Since you're returning this value directly from this function, you shouldn't need the `await` keyword here either. By returning a `Promise.all`, you'll want the thing that calls this function to `await` its result.

Reply:
When I wrap this in `Promise.all` I don't get back a single value, I get back an array which is something like: `['S', 'U', 'C', 'C', 'E', 'S', 'S']` and I haven't found any errors in my testing while doing the await on the reduce. Do you know how to get around this?

Reply:
Ok I finally see your conundrum 😄 

In this case, it might be best to split this into 2 steps:

1) Go do all of the async fetches in a loop
2) Reduce on the returned value

You're right in that you want to reduce to a single value, not the array of values. Because you're executing the `fetch` in the loop, it gets pretty difficult to manage both the async callback and the single value.

Without the `Promise.all`, what you're left with is an array of promises. Take the following example:

```js
const timeout = (duration) => new Promise((resolve, reject) => {
  setTimeout(() => resolve(duration), duration)
})

const test = async () => {
  const value = await [timeout(10), timeout(5), timeout(100)]
  
  console.log(value)
}

test()
```

You'll notice this `await`s the array of promises (which mimic a bit of what you're doing here). In this case, the value that is logged is `Promise`, not the actual resolved value.

By splitting this into 2 steps, it will make it much easier to reduce the single value from the array of fetch results:

```js
const results = await Promise.all(
  contextUids.map(uid => fetch('/url', { method: 'DELETE', ... }).then(res => res.json())
)

return results.reduce((returnCode, { response }) => {
 // ... you get the idea
}, 'SUCCESS')
```

Thoughts?

Reply:
I like that idea! I will implement that

Reply:
Implemented that! let me know if it looks all good :)